keep/providers/splunk_provider/splunk_provider.py

```diff
@@ -0,0 +1,194 @@
+    ):
+        super().__init__(context_manager, provider_id, config)
+
+    def validate_scopes(self) -> dict[str, bool | str]:
+        list_all_objects_scope = "NOT_FOUND"
+        edit_own_object_scope = "NOT_FOUND"
+        try:
+            service = connect(token=self.authentication_config.api_key, host=self.authentication_config.host,
+                              port=self.authentication_config.port)
+            for user in service.users:
+                user_roles = user.content['roles']
+                for role_name in user_roles:
+                    perms = self.__get_role_capabilities(role_name=role_name, service=service)
+                    if not list_all_objects_scope and 'list_all_objects' in perms:
+                        list_all_objects_scope = True
+                    if not edit_own_object_scope and 'edit_own_objects' in perms:
+                        edit_own_object_scope = True
+                    if list_all_objects_scope and edit_own_object_scope:
+                        break
+
+            scopes = {
+                "authenticated": True,
+                "list_all_objects": list_all_objects_scope,
+                "edit_own_objects": edit_own_object_scope,
+            }
+        except Exception as e:
+            self.logger.exception("Error validating scopes")
+            scopes = {
+                "connect_to_client": str(e),
+                "list_all_objects": "UNAUTHENTICATED",
+                "edit_own_objects": "UNAUTHENTICATED",
+            }
+        return scopes
+
+    def validate_config(self):
+        self.authentication_config = SplunkProviderAuthConfig(
+            **self.config.authentication
+        )
+
+    def __get_role_capabilities(self, role_name, service):
+        role = service.roles[role_name]
+        return role.content['capabilities'] + role.content['imported_capabilities']
+
+    def dispose(self):
+        """
+        No need to dispose of anything, so just do nothing.
+        """
+        pass
+
+    def setup_webhook(
+            self, tenant_id: str, keep_api_url: str, api_key: str, setup_alerts: bool = True
+    ):
+        self.logger.info("Setting up Splunk webhook on all Alerts")
+        creation_updation_kwargs = {
+            "actions": "webhook",
+            "action.webhook": "1",
+            "action.webhook.param.url": keep_api_url,
+        }
+        service = connect(token=self.authentication_config.api_key, host=self.authentication_config.host,
+                          port=self.authentication_config.port)
+        for saved_search in service.saved_searches:
+            existing_webhook_url = saved_search["_state"]["content"].get("action.webhook.param.url", None)
+            if existing_webhook_url is None or existing_webhook_url != keep_api_url:
+                saved_search.update(**creation_updation_kwargs).refresh()
+
+    # @staticmethod
+    def _format_alert(self, event: dict) -> AlertDto:
```

Comment:
We still need to figure out what to do here, since this is not straightforward the way the _format_alert mechanism is working today

Reply:
resolved via https://github.com/keephq/keep/pull/950/commits/ec908657343359472bdffc6a150771e41fd56e65